app/models/patient.rb

```diff
@@ -61,7 +61,9 @@ def triage_for_campaign(campaign)
   end
 
   def consent_response_for_campaign(campaign)
-    consent_responses.to_a.find { |t| t.campaign_id == campaign.id }
+    consent_responses.to_a.find do |t|
+      t.campaign_id == campaign.id && t.recorded_at.present?
+    end
```

Comment:
This breaks the state machine spec: `bin/bundle exec rspec ./spec/models/concerns/patient_session_state_machine_concern_spec.rb`